src/Files.App/ViewModels/MainPageViewModel.cs

```diff
@@ -125,15 +125,14 @@ public async Task OnNavigatedToAsync(NavigationEventArgs e)
 					else if (UserSettingsService.GeneralSettingsService.ContinueLastSessionOnStartUp &&
 						UserSettingsService.GeneralSettingsService.LastSessionTabList is not null)
 					{
-						foreach (string tabArgsString in UserSettingsService.GeneralSettingsService.LastSessionTabList)
+						if (AppInstances.Count == 0)
 						{
-							var tabArgs = CustomTabViewItemParameter.Deserialize(tabArgsString);
-							await NavigationHelpers.AddNewTabByParamAsync(tabArgs.InitialPageType, tabArgs.NavigationParameter);
+							foreach (string tabArgsString in UserSettingsService.GeneralSettingsService.LastSessionTabList)
+							{
+								var tabArgs = CustomTabViewItemParameter.Deserialize(tabArgsString);
+								await NavigationHelpers.AddNewTabByParamAsync(tabArgs.InitialPageType, tabArgs.NavigationParameter);
+							}
 						}
-
-						var defaultArg = new CustomTabViewItemParameter() { InitialPageType = typeof(PaneHolderPage), NavigationParameter = "Home" };
-
-						UserSettingsService.GeneralSettingsService.LastSessionTabList = [defaultArg.Serialize()];
```

Comment:
Why this is removed?

Reply:
Because it is no more needed. It used to save the current tabs only when the app exited, but now it saves them every time a tab is opened.

Reply:
I think it's worth keeping for the following scenario

1. Open two windows.
2. Add a few tabs to window _a_.
3. Close window _a_ first, then close window _b_.
4. Upon restarting Files, it will restore the tabs from window _a_, even though window _b_ had the most recent set of tabs.

Reply:
Deleting this code doesn't change that behavior. This code was originally intended to open the home page upon restart after a Files crash.